packages/web-app-external/src/App.vue

```diff
@@ -132,12 +154,56 @@ export default defineComponent({
       }
     }).restartable()
 
+    const determineOpenAsPreview = (appName: string) => {
+      const openAsPreview = configurationManager.options.editor.openAsPreview
+      return (
+        openAsPreview === true || (Array.isArray(openAsPreview) && openAsPreview.includes(appName))
+      )
+    }
+
+    // switch to write mode when edit is clicked
+    const catchClickMicrosoftEdit = (event) => {
+      try {
+        if (JSON.parse(event.data)?.MessageId === 'UI_Edit') {
+          loadAppUrl.perform('write')
+        }
+      } catch (e) {}
+    }
     watch(
-      props.resource,
-      () => {
-        loadAppUrl.perform()
+      applicationName,
+      (newAppName, oldAppName) => {
+        if (determineOpenAsPreview(newAppName) && newAppName !== oldAppName) {
+          window.addEventListener('message', catchClickMicrosoftEdit)
+        } else {
+          window.removeEventListener('message', catchClickMicrosoftEdit)
+        }
+      },
+      {
+        immediate: true
+      }
+    )
+
+    watch(
+      [props.resource],
+      ([newResource], [oldResource]) => {
+        if (isSameResource(newResource, oldResource)) {
+          return
+        }
+
+        debugger
```

Comment:
@dschmidt you forgot to delete it 
<img width="1237" alt="Screenshot 2023-11-16 at 15 55 50" src="https://github.com/owncloud/web/assets/84779829/1a44df7e-5ee9-4f72-be93-9e86e1d6a848">

Reply:
I'll drop it in my PR

Reply:
doh 😂